pkg/controllers/status/cluster_status_controller.go

```diff
@@ -224,6 +238,32 @@ func (c *ClusterStatusController) buildInformerForCluster(cluster *v1alpha1.Clus
 	return singleClusterInformerManager, nil
 }
 
+func (c *ClusterStatusController) initLeaseController(cluster *v1alpha1.Cluster) {
+	// If lease controller has been registered, we skip this function.
+	if _, exists := c.ClusterLeaseControllers.Load(cluster.Name); exists {
+		return
+	}
+
+	// renewInterval is how often the lease renew time is updated.
+	renewInterval := time.Duration(float64(c.ClusterLeaseDuration.Nanoseconds()) * c.ClusterLeaseRenewIntervalFraction)
+
+	nodeLeaseController := lease.NewController(
+		clock.RealClock{},
+		kubeclientset.NewForConfigOrDie(c.Config),
+		cluster.Name,
+		int32(c.ClusterLeaseDuration.Seconds()),
+		nil,
+		renewInterval,
+		util.NamespaceClusterLease,
+		util.SetLeaseOwnerFunc(c.Client, cluster.Name))
```

Comment:
Why use the `c.Client` to retrieve the `Cluster` object again? Can we just use the `cluster` in the parameter?

Reply:
Before lease creating, this function checks whether the cluster can be gotten when trying to set owner ref to the cluster lease. If error happens, e.g. the cluster does not exist, the lease will not be created.